### PR TITLE
Updating worlds after model reorg

### PR DESCRIPTION
**Pull request by quarkytale**
- Bug fix for robot_state_publisher dying, unable to find the model after reorganization.
 ```
Error [SDF.cc:160] Tried to use callback in sdf::findFile(), but the callback is empty.  Did you call sdf::setFindCallback()?
Error Code 13: [/sdf/world[@name="world_demo"]/model[@name="MBARI_WEC_ROS"]/include[0]/uri:/home/anderson/wec/igngzb/openrobotics/workspace/src/buoy_sim/buoy_gazebo/worlds/mbari_wec.sdf:L49]: Msg: Unable to find 
```
- Updating other worlds, launch files and making sure they work as expected.

### CODE REVIEW

**Comment by quarkytale** [comment]
So rviz isn't running as expected robot_state_publisher is looking for mbari_wec model as opposed to mbari_wec_ros, sorry this error slipped from me earlier. I'm still debugging but opening this up if anyone can take a glance.

File: buoy_gazebo/CMakeLists.txt [hunk 1]
```diff
@@ -15,8 +15,9 @@ else()
 endif()
 
 find_package(ament_cmake REQUIRED)
-find_package(rclcpp REQUIRED)
 find_package(buoy_msgs REQUIRED)
+find_package(buoy_description REQUIRED)
```
Comment (chapulina) [inline buoy_gazebo/CMakeLists.txt:19]: We could move this up for alphabetic order

**Comment by andermi** [comment]
Unfortunately, the error is back that I mentioned in #74 , and I can't figure out what's causing it

```
Error:   Could not find the 'robot' element in the xml file
         at line 80 in /tmp/binarydeb/ros-galactic-urdfdom-2.3.5/urdf_parser/src/model.cpp
Failed to parse robot description using: urdf_xml_parser/URDFXMLParser
terminate called after throwing an instance of 'std::runtime_error'
  what():  Unable to initialize urdf::model from robot description
```

**Comment by quarkytale** [comment]
> I actually can't find where the `parser plugin` is actually converting `sdf` to `urdf` referenced by this comment:

When a SDF if published on the `robot_description` topic, `robot_state_publisher` checks in the list of available parsers which can parse the message and uses `sdformat_urdf` from that list (afaik).


This is actually a very inconsistent error. Thank you for opening a bug report, I'll take that forward!

**Comment by quarkytale** [comment]
Some updates on this
- The issue with `ign sdf` callback failing is SDF can't parse the `package://` path. A quick fix is to create and set an `env` variable `SDF_PATH` the same as `IGN_GAZEBO_RESOURCE_PATH`. This seems like a bug I'll take it up.
- With the `Could not find the 'robot' element in the xml file` issue, its essentially the workspace not being sourced or `sdformat_urdf` not installed. Make sure to install deps via `rosdep` too.

**Comment by quarkytale** [comment]
Just pushed a fix, needed to add another path to the `IGN_GAZEBO_RESOURCE_PATH`, solves the rviz problem as well.

**Comment by andermi** [comment]
The tests should pass after I merge https://github.com/osrf/buoy_sim/pull/85 and you merge main into your branch

**Pull request merged**